client/src/components/RelatedProducts/RelatedProducts.jsx

```diff
@@ -20,79 +22,85 @@ const RelatedProducts = ({ productID }) => {
-      const tempData =[];
-      for (var relatedProductId of relatedProducts) {
-       const data = await getProductData(relatedProductId);
-       tempData.push(data);
-      }
-      setProductData(tempData);
+  useEffect(() => {
+    if(relatedProducts.length > 0) {
+      getProductDataAsync();
+      getProductStyleAsync();
     }
+  }, [relatedProducts])
 
-    const getProductData = (relatedProduct) => {
-     return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-sfo/products/${relatedProduct}`, {headers: config})
-        .then((product) => {
-          return product.data;
-        })
-        .catch(() =>{
-          console.log(response.err, 'err at req2')
-        })
+  const getProductDataAsync = async () =>{
+    const tempData =[];
+    for (var relatedProductId of relatedProducts) {
+      const data = await getProductData(relatedProductId);
+      tempData.push(data);
     }
+    setProductData(tempData);
+  }
 
-    console.log('this is product data', productData);
-
-    const getProductStyleAsync = async () => {
-      var tempData = [];
-      for (var relatedProductStylesId of relatedProducts) {
-        const data = await getProductStyle(relatedProductStylesId);
-        tempData.push(data);
-      }
-      console.log(tempData, 'this is tempdata')
-      setProductStyle(tempData);
-    }
+  const getProductData = (relatedProduct) => {
+    return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-sfo/products/${relatedProduct}`, {headers: config})
+      .then((product) => {
+        return product.data;
+      })
+      .catch(() =>{
+        console.log(response.err, 'err at req2')
+      })
+  }
 
-    const getProductStyle = (relatedProduct) => {
-     return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-sfo/products/${relatedProduct}/styles`, {headers: config})
-        .then((product) => {
-          return product.data
-        })
-        .catch(() =>{
-          console.log(response.err, 'err at req3')
-        })
+  const getProductStyleAsync = async () => {
+    var tempData = [];
+    for (var relatedProductStylesId of relatedProducts) {
+      const data = await getProductStyle(relatedProductStylesId);
+      tempData.push(data);
     }
+    setProductStyle(tempData);
+  }
 
-    console.log(productStyle)
-
-    return (
-
+  const getProductStyle = (relatedProduct) => {
+    return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-sfo/products/${relatedProduct}/styles`, {headers: config})
+      .then((product) => {
+        return product.data
+      })
+      .catch(() =>{
+        console.log(response.err, 'err at req3')
+      })
+  }
+
+  console.log('productData', productData);
+  console.log('productStyle', productStyle);
+
+  const CarouselDiv = styled.div`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    border: 3px solid orange;
+    margin: 20px;
+    padding: 10px;
+    height: 350px;
+    width: 100%;
+  `;
```

Comment:
Recommend moving the styling up top outside of the function definition per documentation to avoid drastic slowdown of rendering. 
https://styled-components.com/docs/basics
'Define Styled Components outside of the render method
It is important to define your styled components outside of the render method, otherwise it will be recreated on every single render pass. Defining a styled component within the render method will thwart caching and drastically slow down rendering speed, and should be avoided.'

Reply:
ok thanks i didn't catch i built them within the function, ill go ahead and remove them and place them properly